src/sentry/db/models/manager/base.py

```diff
@@ -263,67 +268,73 @@ def get_from_cache(
         intermediate value.  Callee is responsible for making sure
         the cache key is cleared on save.
         """
-        if not self.cache_fields or len(kwargs) > 1:
+        if not self.cache_fields:
+            raise ValueError("We cannot cache this query. Just hit the database.")
+
+        key, pk_name, value = self._get_cacheable_kv_from_kwargs(kwargs)
+        if key not in self.cache_fields and key != pk_name:
+            raise ValueError("We cannot cache this query. Just hit the database.")
+
+        cache_key = self.__get_lookup_cache_key(**{key: value})
+        local_cache = self._get_local_cache()
+
+        def validate_result(inst: Any) -> M:
+            if isinstance(inst, self.model) and (key != pk_name or int(value) == inst.pk):
+                return inst
+
+            if settings.DEBUG:
+                raise ValueError("Unexpected value type returned from cache")
+            logger.error("Cache response returned invalid value %r", inst)
```

Comment:
```suggestion
            logger.error("Cache response returned invalid value", extra={"instance": inst})
```

If error messages change for each message we'll get terrible grouping.